### PR TITLE
Fix tests. Remove ServiceId and ClusterId getter, turn on file logging

### PR DESCRIPTION
**Pull request by xiazen**
- fix some tests
- remove `ServiceId` and `ClusterId` getter from `TestCluster`, since they are misleading. They only reflect values you set using `TestClusterOptions`, may not reflect values if you configure cluster using legacy `ClusterConfiguration`. 
- turn on file logger for TestCluster by default

### CODE REVIEW

File: src/Orleans.TestingHost/TestClusterBuilder.cs [hunk 1]
```diff
@@ -101,7 +101,7 @@ public TestCluster Build()
             return testCluster;
         }
 
-        private static string CreateClusterId()
+        public static string CreateClusterId()
```
Comment (xiazen) [inline src/Orleans.TestingHost/TestClusterBuilder.cs]: make this public in case user want to configure test cluster id explicitly, through `ClusterConfiguration`

File: src/Orleans.TestingHost/TestClusterHostFactory.cs [hunk 2]
```diff
@@ -227,12 +227,8 @@ private static void TryConfigureTestClusterMembership(IConfiguration configurati
 
         private static void TryConfigureFileLogging(IConfiguration configuration, IServiceCollection services, string name)
         {
-            bool.TryParse(configuration["UseTestClusterMemebership"], out bool configureFileLogging);
```
Comment (xiazen) [inline src/Orleans.TestingHost/TestClusterHostFactory.cs]: not sure why turn on file logging has to depend on `UseTestClusterMemebership`. Seems unrelated. so turn it on by default now.

**Comment by xiazen** [comment]
merged commits from @ReubenBond 's branch for typo fix and use `nameof`

File: src/Orleans.TestingHost/TestClusterHostFactory.cs (hunk 2 above)
Comment (jdom) [reply src/Orleans.TestingHost/TestClusterHostFactory.cs]: I think it was probably a bad copy-paste. this should do `TryParse(configuration[nameof(TestClusterOptions.ConfigureFileLogging)], ...`

---
In reply to: [165226223](https://github.com/dotnet/orleans/pull/3943#discussion_r165226223) [](ancestors = 165226223)

**Comment by xiazen** [comment]
Do not merge yet. want to run it in vso

File: src/Orleans.TestingHost/TestCluster.cs [hunk 3]
```diff
@@ -33,6 +33,7 @@ public class TestCluster
         /// <summary>
         /// Primary silo handle
         /// </summary>
+        /// <remarks>This only provide valid data if your cluster are using GrainBasedMemebership</remarks>
```
Comment (ReubenBond) [inline src/Orleans.TestingHost/TestCluster.cs]: `This handle is valid only when using Grain-based membership`

**Comment by ReubenBond** [comment]
Will merge when VSO CI completes

**Comment by jdom** [comment]
@ReubenBond FYI, Xiao is going to address the ConfigureFileLogging comment

**Comment by ReubenBond** [comment]
ok, we can leave this until tomorrow then

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

**Comment by xiazen** [comment]
run a vso build on this : 
https://orleans.visualstudio.com/Orleans/_build/index?buildId=7848&_a=summary&tab=ms.vss-test-web.test-result-details

there's still remaining test failure related to 

>  Lifecycle start canceled due to errors at stage 4000\r\n---- System.TimeoutException : WithTimeout has timed out after 00:05:00.

We can fix those bugs in another pr. this pr fixed some tests failures and did improvement on TestCluster. It can be merged by itself.

**Comment by jdom** [comment]
I agree, they can be fixed separately. BTW, we should make that exception clearer, not just fix the tests so that it doesn't happen.
That exception is extremely cryptic, and you can't know what went wrong during the lifecycle calls.

**Comment by jdom** [comment]
BTW, these are new and related failures (to the changes in this PR), right? https://ci.dot.net/job/dotnet_orleans/job/master/job/functional_prtest/1115/
